src/components/data-table/ha-data-table.ts

```diff
@@ -543,6 +553,10 @@ export class HaDataTable extends BaseElement {
         text-align: left;
       }
 
+      .mdc-data-table__header-cell--icon {
```

Comment:
Can we make the text color secondary text color ? Black draws too much attention. The other text should be `--primary-text-color` if it isn't yet.